### PR TITLE
Proposer: aggregated unaggregated attestations for block

### PR DESCRIPTION
**Pull request by terencechain**
This PR ensures that unaggregated attestations in the block are aggregated

The unaggregated attestations are aggregated with the aggregated attestations then sort by higher slot then by higher aggregation bit count. Finally, It's capped at `MAX_ATTESTATIONS` length

### CODE REVIEW

File: beacon-chain/rpc/validator/proposer.go [hunk 1]
```diff
@@ -664,9 +665,15 @@ func (vs *Server) packAttestations(ctx context.Context, latestState *stateTrie.B
...
+		uAtts, err = attaggregation.Aggregate(uAtts)
+		if err != nil {
+			return nil, err
+		}
+
 		if numUAtts+numAtts > params.BeaconConfig().MaxAttestations {
 			uAtts = uAtts[:params.BeaconConfig().MaxAttestations-numAtts]
 		}
+
 		atts = append(atts, uAtts...)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator/proposer.go:677]:
```suggestion
		uAtts, err = attaggregation.Aggregate(uAtts)
		if err != nil {
			return nil, err
		}

		atts = append(atts, uAtts...)
		atts, err = attaggregation.Aggregate(atts)
		if err != nil {
		   return nil, err
		}
		if len(atts) > params.BeaconConfig().MaxAttestations {
		   atts = sort.MostProfitable(atts)
		   atts = atts[:params.BeaconConfig().MaxAttestations]
		}
		atts = sort.MostProfitable(atts)
```

What about something like this?

Comment (terencechain) [reply beacon-chain/rpc/validator/proposer.go:677]: Much better

Comment (prestonvanloon) [reply beacon-chain/rpc/validator/proposer.go]: The sort thing is not implemented though!

File: beacon-chain/rpc/validator/proposer.go [hunk 2]
```diff
@@ -664,10 +665,16 @@ func (vs *Server) packAttestations(ctx context.Context, latestState *stateTrie.B
...
+		atts, err = attaggregation.Aggregate(atts)
+		if err != nil {
+		   return nil, err
+		}
+		if len(atts) > params.BeaconConfig().MaxAttestations {
+		   atts = sort.MostProfitable(atts)
+		   atts = atts[:params.BeaconConfig().MaxAttestations]
+		}
+		atts = sort.MostProfitable(atts)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator/proposer.go:677]:
```suggestion
```

I messed up my original suggestoin. only need to sort once

File: go.sum [hunk 3]
```diff
@@ -950,8 +950,6 @@ github.com/prometheus/client_golang v0.9.1/go.mod h1:7SWBe2y4D6OKWSNQJUaRYU/AaXP
 github.com/prometheus/client_golang v1.0.0/go.mod h1:db9x61etRT2tGnBNRi70OPL5FsnadC4Ky3P0J6CfImo=
 github.com/prometheus/client_golang v1.4.1/go.mod h1:e9GMxYsXl05ICDXkRhurwBS4Q3OK1iX/F2sw+iXX5zU=
 github.com/prometheus/client_golang v1.5.1/go.mod h1:e9GMxYsXl05ICDXkRhurwBS4Q3OK1iX/F2sw+iXX5zU=
-github.com/prometheus/client_golang v1.6.0 h1:YVPodQOcK15POxhgARIvnDRVpLcuK8mglnMrWfyrw6A=
```
Comment (terencechain) [inline go.sum:953]: Go mod tidy check was red. I had to do clean this up

**Pull request merged**